clients/src/main/java/org/apache/kafka/common/TopicCollection.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common;
+
+import java.util.Collection;
+
+/**
+ * A class used to represent a collection of topics. This collection may define topics by topic name
+ * attribute or topic ID attribute. Subclassing this class beyond the classes provided here is not supported.
+ */
+public abstract class TopicCollection {
+    private final TopicAttribute attribute;
+
+    /**
+     * An enum used to describe how topics in the collection are identified
+     */
+    public enum TopicAttribute {
```

Comment:
I kept this TopicAttribute field for now, though we could case on the class (TopicIdCollection/TopicNameCollection) One reason I didn't remove it is that the result class(es) still needs it. (Since we are building off the existing class). If there is a way to remove from the result class, or a better place to put the attribute, let me know.